### ISSUE TITLE
When extending tailwind.config.js with new colors then applying those colors to a Flowbite component that already has the newly added/ extended property, the default supersedes customization

### ISSUE THREAD

**Issue by cududa** [themesberg/flowbite-react#530]
**Describe the bug**
When using the Flowbite header component (which by default is styled as `bg-white`), adding a custom color to tailwind.config, then applying the color to the component, the Flowbite component doesn't respect the custom color and overwrites your new color

**To Reproduce**

1. Create a Nav.js component using the Flowbite **Navbar with dropdown** from https://flowbite-react.com/navbars
2. In your tailwind.css.config file, extend it like so:
```
/** @type {import('tailwindcss').Config} */
module.exports = {
	content: [
		'./src/**/*.{js,jsx,ts,tsx}',
		'node_modules/flowbite-react/**/*.{js,jsx,ts,tsx}'
	],
	theme: {
		extend: {
			colors: {
				mycolors: {
					light: '#031017',
					dark: '#2a2c3d',
					bright: '#3dbaff',
					pale: '#4b5563',
					white: '#fff'
				}
			}
		}
	},
	plugins: [require('flowbite/plugin')]
};
```
3. In your Nav.js component, go to the Flowbite `<Navbar>` configuration. Add className="bg-mycolors-dark"
5. The default bg-white supersedes the custom color. See screenshot.
6. Removing bg-white in dev tools allows the custom color to appear

**Expected behavior**
Components should behave as all other Tailwind/ JSX components do - whereby the user applied customization takes precedence over the component's default. 

**Screenshots**

1. <img width="1680" alt="image" src="https://user-images.githubusercontent.com/6423489/211928051-7301c150-f7b0-4690-a9a9-3de74e921353.png">
2. Remove bg-white
<img width="1680" alt="image" src="https://user-images.githubusercontent.com/6423489/211928141-72d631bd-0d48-4f4c-9773-9a9950718790.png">

**Project information:**
- Tailwind: 3.2.4
- Flowbite: 1.6.0
- Flowbite React: 0.3.7
- Type: CRA

**Additional context**

**Comment by rluders** [comment]
> TLDR: `className` only adds classes to the existing theme. In order to overwrite any theme style you need to prefix the classes with `!`, or just create your own flowbite-react theme.

@cududa could you try to use `className="!bg-mycolors-dark"` and verify if the issue continue? BTW, if you are looking for a more reliable way to customize things I do recommend you take a look into flowbite-react theme system: https://flowbite-react.com/theme

We still need to improve the documentation about it, but it should be done after the `0.4.0` release, 'cause we are improving the theme system to allow overwriting the theme using theme-per-component approach.

**Comment by rluders** [comment]
I tested it yesterday and I wasn't able to confirm the issue. I'm closing this one.
